arch/arm/core/aarch32/userspace.S

```diff
@@ -574,18 +553,21 @@ dispatch_syscall:
     orrs r2, r2, r3
     msr CONTROL, r2
     pop {r2, r3}
-#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE) \
+    || defined(CONFIG_CPU_CORTEX_R)
```

Comment:
Could this instance of CONFIG_CPU_CORTEX_R be replaced by CONFIG_ARMV7_R as well?